ctru-rs/src/mii.rs

```diff
@@ -0,0 +1,479 @@
+        ];
+        let raw_details: [bool; 16] =
+            get_and_concat_vec_bit(raw_mii_data.as_slice(), &[0x18, 0x19])
+                .try_into()
+                .unwrap();
+        let raw_utf16_name = &raw_mii_data.as_slice()[0x1A..0x2D];
+        let height = raw_mii_data[0x2E];
+        let width = raw_mii_data[0x2F];
+        let raw_face_style = vec_bit(raw_mii_data[0x30]);
+        let raw_face_details = vec_bit(raw_mii_data[0x31]);
+        let hair_style = raw_mii_data[0x32];
+        let raw_hair_details = vec_bit(raw_mii_data[0x33]);
+        let raw_eye_details: [bool; 32] =
+            get_and_concat_vec_bit(raw_mii_data.as_slice(), &[0x34, 0x35, 0x36, 0x37])
+                .try_into()
+                .unwrap();
+        let raw_eyebrow_details: [bool; 32] =
+            get_and_concat_vec_bit(raw_mii_data.as_slice(), &[0x38, 0x39, 0x3A, 0x3B])
+                .try_into()
+                .unwrap();
+        let raw_nose_details: [bool; 16] =
+            get_and_concat_vec_bit(raw_mii_data.as_slice(), &[0x3C, 0x3D])
+                .try_into()
+                .unwrap();
+        let raw_mouth_details: [bool; 16] =
+            get_and_concat_vec_bit(raw_mii_data.as_slice(), &[0x3E, 0x3F])
+                .try_into()
+                .unwrap();
+        let raw_mustache_details: [bool; 16] =
+            get_and_concat_vec_bit(raw_mii_data.as_slice(), &[0x40, 0x41])
+                .try_into()
+                .unwrap();
+        let raw_beard_details: [bool; 16] =
+            get_and_concat_vec_bit(raw_mii_data.as_slice(), &[0x42, 0x42])
+                .try_into()
+                .unwrap();
+        let raw_glass_details: [bool; 16] =
+            get_and_concat_vec_bit(raw_mii_data.as_slice(), &[0x44, 0x45])
+                .try_into()
+                .unwrap();
+        let raw_mole_details: [bool; 16] =
+            get_and_concat_vec_bit(raw_mii_data.as_slice(), &[0x46, 0x47])
+                .try_into()
+                .unwrap();
+        let raw_utf16_author = &raw_mii_data.as_slice()[0x48..0x5C];
+
+        let mii_name = utf16_byte_pairs_to_string(raw_utf16_name);
+        let author_name = utf16_byte_pairs_to_string(raw_utf16_author);
+
+        let options = MiiDataOptions {
+            is_copying_allowed: *raw_options.first().unwrap(),
+            is_profanity_flag_enabled: *raw_options.get(1).unwrap(),
+            region_lock: {
+                let first_bit = raw_options[3];
+                let second_bit = raw_options[2];
+                if !first_bit && !second_bit {
+                    // 0b00
+                    RegionLock::None
+                } else if !first_bit && second_bit {
+                    // 0b01
+                    RegionLock::Japan
+                } else if first_bit && !second_bit {
+                    // 0b10
+                    RegionLock::USA
+                } else {
+                    RegionLock::Europe
+                }
```

Comment:
Do this refactoring for the other similar instances below as well.
```suggestion
                match (raw_options[3], raw_options[2]) {
                    (false, false) => RegionLock::None,
                    (false, true) => RegionLock::Japan,
                    (true, false) => RegionLock::USA,
                    (true, true) => RegionLock::Europe,
                }
```

Reply:
Ohhh that's actually great! I hadn't thought of doing that